hive/src/test/java/org/apache/iceberg/hive/legacy/TestLegacyHiveTableScan.java

```diff
@@ -93,44 +97,53 @@ public void testHiveScanUnpartitioned() throws Exception {
   public void testHiveScanSinglePartition() throws Exception {
     String tableName = "single_partition";
     Table table = createTable(tableName, DATA_COLUMNS, PARTITION_COLUMNS);
-    addPartition(table, ImmutableList.of("1"), AVRO, "A", "B");
-    filesMatch(ImmutableMap.of("pcol=1/A", AVRO, "pcol=1/B", AVRO), hiveScan(table));
+    addPartition(table, ImmutableList.of("ds", 1), AVRO, "A", "B");
+    filesMatch(ImmutableMap.of("pcol=ds/pIntCol=1/B", AVRO, "pcol=ds/pIntCol=1/A", AVRO), hiveScan(table));
   }
 
   @Test
   public void testHiveScanMultiPartition() throws Exception {
     String tableName = "multi_partition";
     Table table = createTable(tableName, DATA_COLUMNS, PARTITION_COLUMNS);
-    addPartition(table, ImmutableList.of("1"), AVRO, "A");
-    addPartition(table, ImmutableList.of("2"), AVRO, "B");
-    filesMatch(ImmutableMap.of("pcol=1/A", AVRO, "pcol=2/B", AVRO), hiveScan(table));
+    addPartition(table, ImmutableList.of("ds", 1), AVRO, "A");
+    addPartition(table, ImmutableList.of("ds", 2), AVRO, "B");
+    filesMatch(ImmutableMap.of("pcol=ds/pIntCol=2/B", AVRO, "pcol=ds/pIntCol=1/A", AVRO), hiveScan(table));
   }
 
   @Test
   public void testHiveScanMultiPartitionWithFilter() throws Exception {
     String tableName = "multi_partition_with_filter";
     Table table = createTable(tableName, DATA_COLUMNS, PARTITION_COLUMNS);
-    addPartition(table, ImmutableList.of("1"), AVRO, "A");
-    addPartition(table, ImmutableList.of("2"), AVRO, "B");
-    filesMatch(ImmutableMap.of("pcol=2/B", AVRO), hiveScan(table, Expressions.equal("pcol", "2")));
+    addPartition(table, ImmutableList.of("ds", 1), AVRO, "A");
+    addPartition(table, ImmutableList.of("ds", 2), AVRO, "B");
+    filesMatch(
+        ImmutableMap.of("pcol=ds/pIntCol=1/A", AVRO, "pcol=ds/pIntCol=2/B", AVRO),
+        hiveScan(table, Expressions.equal("pcol", "ds")));
+
+    AssertHelpers.assertThrows(
```

Comment:
Non string filtering is only supported for direct sql and direct sql only works for mysql backend.

Reply:
Can you split this failure into a different test please?